storage.js

```diff
@@ -33,7 +35,82 @@ module.exports = {
 	 * @return	{Object} Video object, but it may contain missing properties.
 	 */
 	getVideoInfo(service, id) {
-		// TODO: get video info from db
-		return new Promise(resolve => resolve({ service, id }));
+		return CachedVideo.findOne({ where: { service: service, serviceId: id } }).then(cachedVideo => {
+			if (cachedVideo === null) {
+				console.log("Cache missed:", service, id);
+				return { service, id };
+			}
+			const origCreatedAt = moment(cachedVideo.createdAt);
+			const lastUpdatedAt = moment(cachedVideo.updatedAt);
+			const today = moment();
+			// We check for changes every at an interval of 30 days, unless the original cache date was
+			// less than 7 days ago, then the interval is 7 days. The reason for this is that the uploader
+			// is unlikely to change the video info after a week of the original upload. Since we don't store
+			// the upload date, we pretend the original cache date is the upload date. This is potentially an
+			// over optimization.
+			const isCachedInfoValid = lastUpdatedAt.diff(today, "days") <= (origCreatedAt.diff(today, "days") <= 7) ? 7 : 30;
+			let video = {
+				service: cachedVideo.service,
+				id: cachedVideo.serviceId,
+			};
+			// We only invalidate the title and description because those are the only ones that can change.
```

Comment:
The thumbnail can also change. The video author may also unpublish a video; might need to test what the API returns in this case.

Reply:
The thumbnail may change, but the _url_ to the thumbnail doesn't (needs confirmation).
For example, this video: https://www.youtube.com/watch?v=mOVOS9AjgFs
Has thumbnail: https://img.youtube.com/vi/mOVOS9AjgFs/hqdefault.jpg